File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
# similar to fibonacci t0= 0, t1= 1, t2= 1 and t3= t1+t2+t3
        if n == 0:
            return 0
        elif n < 3:
            return 1
        else:
            n1, n2, n3 = 0, 1, 1
            for i in range(3, n+1):
                temp = n1 + n2 + n3
                n1, n2, n3 = n2, n3, temp
            return temp
# return nth tribonacci

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
// similar to fibo if n value < 2 then return 1 and if n = 0 then return 0
        if(n == 0)
            return 0;
        else if (n < 3)
            return 1;

        else{
// initialize n0, n1, n2
            int n0 = 0, n1 = 1, n2 = 1;
            for (int i=3; i<=n;i++){
// Tn+3 = Tn + Tn+1 + Tn+2
                int temp = n0 + n1 + n2;
// assign the new values of n0, n1, n2
                n0 = n1;
                n1 = n2;
                n2 = temp;
            }
// return the value for n3 
            return n2;
        }
        
    }
}